### PR TITLE
pull cloning all parameterers out to a seperate method

### PR DESCRIPTION
**Pull request by karlnapf**

### CODE REVIEW

File: src/shogun/base/SGObject.cpp [hunk 1]
```diff
@@ -758,21 +756,39 @@ CSGObject* CSGObject::clone()
...
+bool CSGObject::clone_parameters(CSGObject* other)
+{
+	REQUIRE(other, "Provided instance must be non-empty.\n");
+	REQUIRE(other->m_parameters->get_num_parameters() == m_parameters->get_num_parameters(),
+		"Number of parameters of provided instance (%d) must match this instance (%d).\n",
+		other->m_parameters->get_num_parameters(), m_parameters->get_num_parameters());
+	REQUIRE(!strcmp(other->get_name(), get_name()),
+		"Name of provided instance (%s) must match this instance (%d).\n",
+		other->get_name(), get_name());
+
```
Comment (lkuchenb) [inline src/shogun/base/SGObject.cpp]: Trailing whitespace

File: src/shogun/base/SGObject.cpp [hunk 2]
```diff
@@ -758,21 +756,39 @@ CSGObject* CSGObject::clone()
...
+				" into parameter \"%s\" of this instance %s failed.\n",
+				other->m_parameters->get_parameter(i)->m_name, i,
+				other->get_name(), m_parameters->get_parameter(i)->m_name);
+			return false;
 		}
 	}
-
-	SG_DEBUG("leaving %s::clone(): Clone successful\n", get_name());
-	return copy;
+
```
Comment (lkuchenb) [inline src/shogun/base/SGObject.cpp]: Trailing whitespace

File: src/shogun/base/SGObject.cpp [hunk 3]
```diff
@@ -758,21 +756,39 @@ CSGObject* CSGObject::clone()
...
+	return copy;
+}
+
+bool CSGObject::clone_parameters(CSGObject* other)
+{
+	REQUIRE(other, "Provided instance must be non-empty.\n");
+	REQUIRE(other->m_parameters->get_num_parameters() == m_parameters->get_num_parameters(),
+		"Number of parameters of provided instance (%d) must match this instance (%d).\n",
+		other->m_parameters->get_num_parameters(), m_parameters->get_num_parameters());
+	REQUIRE(!strcmp(other->get_name(), get_name()),
```
Comment (lkuchenb) [inline src/shogun/base/SGObject.cpp]: Wrong format specifier %d

**Comment by karlnapf** [comment]
Travis error unrelated, merging

**Pull request merged**